src/main/java/org/jitsi/jigasi/transcription/Participant.java

```diff
@@ -190,7 +207,90 @@ public String getAvatarUrl()
             return null;
         }
 
-        return ((ChatRoomMemberJabberImpl) chatMember).getAvatarUrl();
+        ChatRoomMemberJabberImpl memberJabber
+            = ((ChatRoomMemberJabberImpl) this.chatMember);
+
+        String url;
+        if((url = (String) memberJabber.getData(
+            ChatRoomMemberPresenceExtensionReader
+                .IDENTITY_AVATAR_URL)) != null)
+        {
+            return url;
+        }
+        else if((url = memberJabber.getAvatarUrl()) != null)
+        {
+            return url;
+        }
+
+        String email;
+        if((email = getEmail()) != null)
+        {
+            return String.format(GRAVARAR_URL_FORMAT,
+                Util.stringToMD5hash(email));
+        }
+
+        // Create a nice looking meeple avatar when avatar-url nor email is set
+        String avatarId;
+        if((avatarId
+            = (String) memberJabber.getData(
+            ChatRoomMemberPresenceExtensionReader.AVATAR_ID)) != null)
+        {
+            return String.format(MEEPLE_URL_FORMAT,
+                Util.stringToMD5hash(avatarId));
+        }
+
+        return null;
```

Comment:
Perhaps we can still return a valid random avatar based on the user id if everything else is missing?